src/map/lua/lua_baseentity.cpp

```diff
@@ -16631,6 +16631,19 @@ void CLuaBaseEntity::useMobAbility(sol::variadic_args va)
     // clang-format on
 }
 
+/************************************************************************
+ *  Function: getAbilityDistance()
+ *  Purpose : Returns the distance for a specified ability from mob_skills
+ *  Example : mob:getAbilityDistance(740)
+ *  Notes   : Uses Ability ID Only
+ ************************************************************************/
+float CLuaBaseEntity::getAbilityDistance(uint16 skillID)
+{
+    auto* PMobSkill{ battleutils::GetMobSkill(skillID) };
+
+    return PMobSkill->getDistance();
```

Comment:
Need to check null `PMobSkill` here

Reply:
Good catch.  Added a check for null and just returns silently
```cpp
        auto defaultTarget = mobObj->GetBattleTarget();

        if (defaultTarget == nullptr)
        {
            return;
        }
        battleutils::DrawIn(defaultTarget, mobObj, mobObj->GetMeleeRange() - 0.2f);
        return;
```

Reply:
Added a check for null here that shows a warning about invalid skillID being specified and return `0.0f`
```cpp
    if (PMobSkill == nullptr)
    {
        ShowWarning("Invalid SkillID (%d) returned null, returning 0 for distance.", skillID);
        return 0.0f;
    }
```